sunpy/net/jsoc/jsoc.py

```diff
@@ -489,165 +497,131 @@ def _process_time(self, time):
         """
-        keywords = ['DATE', 'TELESCOP', 'INSTRUME', 'T_OBS', 'WAVELNTH',
-                    'WAVEUNIT']
+
+        keywords_default = ['DATE', 'TELESCOP', 'INSTRUME', 'T_OBS', 'WAVELNTH']
+        isMeta = iargs.get('meta', False)
+        c = drms.Client()
+
+        if isMeta:
+            keywords = '***ALL***'
+        else:
+            keywords = iargs.get('keys', keywords_default)
 
         if not all([k in iargs for k in ('start_time', 'end_time', 'series')]):
             error_message = "Both Time and Series must be specified for a "\
                             "JSOC Query"
             raise ValueError(error_message)
 
+        if not isinstance(keywords, list) and not isinstance(keywords, six.string_types):
+            error_message = "Keywords can only be passed as a list or"\
+                            " comma-separated strings."
+            raise ValueError(error_message)
+
+        segments = iargs.get('segment', '')
+        if segments:
+            if not isinstance(segments, list) and not isinstance(segments, six.string_types):
+                error_message = "Segments can only be passed as a list or"\
+                                " comma-separated strings."
+                raise ValueError(error_message)
+
+        pkeys = c.pkeys(iargs['series'])
+        pkeys_passed = iargs.get('primekeys', None)
+        if pkeys_passed is not None:
+            if not set(list(pkeys_passed.keys())) < set(pkeys):
+                error_message = "Unexpected PrimeKeys were passed. The series {series} "\
+                                "supports the following PrimeKeys {pkeys}"
+                raise TypeError(error_message.format(series=iargs['series'], pkeys=pkeys))
+
+        wavelength = iargs.get('wavelength', '')
+        if wavelength:
+            if 'WAVELNTH' not in pkeys:
+                error_message = "The series {series} does not support wavelength attribute."\
+                                " Following primekeys are supported {pkeys}"
+                raise TypeError(error_message.format(series=iargs['series'], pkeys=pkeys))
+
+        si = c.info(iargs['series'])
+        segs = list(si.segments.index.values)
+        segs_passed = iargs.get('segment', None)
+        if segs_passed is not None:
+            if not set(segs_passed) < set(segs):
+                error_message = "Unexpected Segments were passed. The series {series} "\
+                                "contains the following Segments {segs}"
+                raise TypeError(error_message.format(series=iargs['series'], segs=segs))
+
+        iargs['start_time'] = self._process_time(iargs['start_time'])
+        iargs['end_time'] = self._process_time(iargs['end_time'])
+
         postthis = {'ds': self._make_recordset(**iargs),
-                    'op': 'rs_list',
                     'key': str(keywords)[1:-1].replace(' ', '').replace("'", ''),
                     'seg': '**NONE**',
```

Comment:
`posthis` no more goes into JSON now. It is just a dict to store values. `seg` has stay here, I have to still a small change to make regarding this.